src/opnsense/scripts/firmware/repos/SunnyValley.php

```diff
@@ -0,0 +1,49 @@
+#!/usr/local/bin/php
+<?php
+
+/*
+ * Copyright (C) 2023 Franco Fichtner <franco@opnsense.org>
+ * Copyright (C) 2024 Hasan Ucak <hasan@sunnyvalley.io>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+require_once 'util.inc';
+
+$conf = '/usr/local/etc/pkg/repos/SunnyValley.conf';
+$uuidPattern = '/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/';
+
+/* ensure that SunnyValley repo is off to avoid obvious breakage */
+# @copy($conf . '.shadow', $conf);
+if (file_exists($conf) && file_exists('/usr/local/zenarmor/bin/eastpect')) {
+    $node_uuid = shell_safe('/usr/local/zenarmor/bin/eastpect -s');
+    if ($node_uuid != '') {
+        $fileContents = file_get_contents($conf);
```

Comment:
looks promising, but I would recommend using the copy of the unmodified file to replace it in on direction only.. hoping the uuid is there 

```suggestion
        $fileContents = file_get_contents($conf . '.shadow');
```

it avoids the else case

Reply:
The code has been updated with your review.